doc/tutorials/Chapter_2/00-Basic_Logic.md

```diff
@@ -0,0 +1,534 @@
+answer    <=  mux(selectA, a, b) // answer = selectA ? a : b
+```
+
+Great, now that you've learned all about our operators, let's continue our journey and create an `AND` gate. 
+
+## Logic Gate: Part 2
+
+We can use the `&` operator that we learned earlier to create an `AND` logic gate.
+
+```dart
+class LogicGate extends Module {
+  
+  // TODO(user): (Optional) Public attributes can register here.
+  // -----------------------------------------------------------
+  late final Logic a;
+  late final Logic b;
+  late final Logic c;
+
+  LogicGate() : super(name: 'LogicGate') {
+    // TODO(user): (Required) Paste your Logic initialization here.
+    // ------------------------------------------------------------
+    // Create input and output signals
+    a = Logic(name: 'input_a');
+    b = Logic(name: 'input_b');
+    c = Logic(name: 'output_c');
+
+    // TODO(user): (Required) Declare your input and output port.
+    // ----------------------------------------------------------
+    // Add ports
+    final signal1 = addInput('input_a', a, width: a.width);
+    final signal2 = addInput('input_b', b, width: b.width);
+    final signal3 = addOutput('output_c', width: c.width);
+
+    // TODO(user): (Optional) Logic Operations.
+    // ----------------------------------------
+
+    // Note that the & symbols is the AND operator
+    c <= signal1 & signal2;
+    signal3 <= c;
+  }
+}
+```
+
+### Exercise 2:
+
+1. Can you optimize the code? Can you remove the redundance from the code? 
+
+You can find the executable code at [e_logic_gate_part_2.dart](./e_logic_gate_part_2.dart). Congratulations! You have created your logic gate. Let's move on to the next section to test our gate.
+
+### Non-synthesizable signal deposition (put)
+
+Do you still remember the `put()` function that was used in the previous section? It is used to send a simulated signal to the input `Logic`.
+
+For testbench code or other non-synthesizable code, you can use `put` on any `Logic` to deposit a value on the signal.
+
+Now let's see an example of how to deposit signals for testing using `put`. At this point, you should be familiar with how you can deposit signals.
+
+Note: This section of code **does not need to run**. 
+
+```dart
+b = Logic(width:4);
+
+// you can put an integer directly into a signal.
+a.put(4);
+```
+
+## Logic Gate: Part 3
+
+Now, we can test our logic gate with the simulator.
+
+```dart
+// ...
+
+void main() async {
+  // ...
+
+  // TODO(user): (Optional) Simulate your Module.
+  // --------------------------------------------
+
+  // Let build a truth table
+  int portC;
+  print('\nBuild Truth Table: ');
+  for (var i = 0; i <= 1; i++) {
+    for (var j = 0; j <= 1; j++) {
+      basicLogic.a.put(i);
+      basicLogic.b.put(j);
+      // print('a: $i, b: $j c: ${basicLogic.c.value.toInt()}');
+      portC = basicLogic.signals
```

Comment:
this search using `firstWhere` seems unnecessary for the tutorial?  Why not just refer to the output?  Maybe related to previous comment about whether we should even use `Module` here